e2e/LongHaul/amqpPooling/IotHub.cs

```diff
@@ -0,0 +1,183 @@
+        private readonly IotHubClientTransportSettings _transportSettings;
+        private readonly IotHubClientOptions _clientOptions;
+        private static IList<Device> s_devices;
+        private static IDictionary<string, DeviceOperations> s_deviceOperations;
+
+        private static readonly TimeSpan s_messageLoopSleepTime = TimeSpan.FromSeconds(3);
+        private readonly ConcurrentQueue<SystemHealthTelemetry> _messagesToSend = new();
+
+        private SemaphoreSlim _lifetimeControl = new(1, 1);
+
+        public IotHub(Logger logger, Parameters parameters, IList<Device> devices)
+        {
+            _logger = logger;
+            _hubConnectionString = parameters.IotHubConnectionString;
+            _transportSettings = parameters.GetTransportSettingsWithPooling();
+            _clientOptions = new IotHubClientOptions(_transportSettings);
+
+            s_devices = devices;
+            s_deviceOperations = new Dictionary<string, DeviceOperations>();
+        }
+
+        public Dictionary<string, string> TelemetryUserProperties { get; } = new();
+
+        public async Task InitializeAsync()
+        {
+            await _lifetimeControl.WaitAsync().ConfigureAwait(false);
+
+            var helper = new IotHubConnectionStringHelper(_hubConnectionString);
+
+            try
+            {
+                _logger.Trace(
+                    $"Creating {s_devices.Count} device clients with transport settings [{_transportSettings.ToString()}].",
+                    TraceSeverity.Information);
+
+                foreach (var device in s_devices)
+                {
+                    string deviceConnectionString = $"HostName={helper.HostName};DeviceId={device.Id};SharedAccessKey={device.Authentication.SymmetricKey.PrimaryKey}";
+
+                    var deviceClient = new IotHubDeviceClient(deviceConnectionString, _clientOptions);
+
+                    // Set the device client connection status change handler
+                    var amqpConnectionStatusChange = new AmqpConnectionStatusChange(_logger.Clone());
+                    deviceClient.ConnectionStatusChangeCallback = amqpConnectionStatusChange.ConnectionStatusChangeHandler;
+
+                    await deviceClient.OpenAsync().ConfigureAwait(false);
+
+                    var deviceOperations = new DeviceOperations(deviceClient, device.Id, _logger.Clone());
+                    s_deviceOperations.Add(device.Id, deviceOperations);
+                }
+            }
+            finally
+            {
+                _lifetimeControl.Release();
+            }
+        }
+
+        public async Task SendTelemetryMessagesAsync(CancellationToken ct)
+        {
+            while (!ct.IsCancellationRequested)
+            {
+                await Task.Delay(s_messageLoopSleepTime, ct).ConfigureAwait(false);
+
+                List<Task> deviceOperationTasks = new();
+
+                _messagesToSend.TryDequeue(out SystemHealthTelemetry systemHealth);
```

Comment:
We used a queue for message sending in the normal long haul. I'd be fine if in this one you simply sleep/send telemetry from the device, so no need for queue.

This would work because we can collect system performance metrics and log them with AI in a loop that does not get paused by disconnections.